### PR TITLE
output: indicate when pipeline errored

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @jazzdan, @yuindustries,

Please review the following commits I made in branch maiamcc/ch136/good-err-output:

2fbd65bf3b7de7277a884b2650e8da23d504b8fa (2018-09-07 11:35:36 -0400)
output: indicate when pipeline errored

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by jazzdan** [comment]
Can you leave a little sample output here?

File: internal/engine/image_build_and_deployer.go [hunk 1]
```diff
@@ -33,15 +33,15 @@ func NewImageBuildAndDeployer(b build.ImageBuilder, k8sClient k8s.Client, histor
 	}, nil
 }
 
-func (ibd ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, service model.Service, state BuildState) (BuildResult, error) {
+func (ibd ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, service model.Service, state BuildState) (br BuildResult, err error) {
```
Comment (jazzdan) [inline internal/engine/image_build_and_deployer.go]: Why are we using named returns here now?

**Comment by maiamcc** [comment]
screen shot 2018-09-06 at 6 30 38 pm

^ sample output!

File: internal/engine/image_build_and_deployer.go (hunk 1 above)
Comment (maiamcc) [reply internal/engine/image_build_and_deployer.go]: cuz we want the deferred call to `EndPipeline` to report on the state of the error that the function is returning. without named returns, we have no way to latch onto the value we're returning, and rely on it having been assigned to a variable beforehand (and not just any variable, but a specific one -- so users can't name multiple errors different things).
```
func Stuff() error {
    var err error
    defer EndPipeline(err)
    ...
    return fmt.Errorf("this error isn't bound to 'err'"!)
```
^ if we don't use named returns, we have no way of grabbing that error we return here.

**Comment by maiamcc** [comment]
gonna land but @dbentley could you also take a look and lmk if i did anything silly?

**Pull request merged**